includes/class-wc-checkout.php

```diff
@@ -271,7 +271,7 @@ public function get_checkout_fields( $fieldset = '' ) {
 			// Add accessibility labels to fields that have placeholders.
 			foreach ( $fields as $single_field_type => $field ) {
 				if ( empty( $field['label'] ) && ! empty( $field['placeholder'] ) ) {
-					$this->fields[ $field_type ][ $single_field_type ]['label']       = $field['placeholder'];
+					$this->fields[ $field_type ][ $single_field_type ]['label'] = ( 'billing_address_2' === $single_field_type && 'optional' === get_option( 'woocommerce_checkout_address_2_field', 'optional' ) ) ? __( 'Apartment, suite, unit etc.', 'woocommerce' ) : $field['placeholder'];
```

Comment:
I think it would be better to change this string at its source in: https://github.com/woocommerce/woocommerce/blob/d3a33f8d0547b709b4c7d7753b209802e85c3f44/includes/class-wc-countries.php#L650

Since the "optional" span isn't added unless the field isn't required, that should probably be all that needs done?